src/SwedbankPay.Sdk/SwedbankPayHttpClient.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            this.client = client;
+            this.logger = logger;
+        }
+
+
+        internal async Task<string> GetRaw(string url)
+        {
+            var msg = new HttpRequestMessage(HttpMethod.Get, url);
+            msg.Headers.Add("Accept", "application/json");
+
+            HttpResponseMessage response;
+            try
+            {
+                response = await this.client.SendAsync(msg);
+            }
+            catch (HttpRequestException e)
+            {
+                throw new BadRequestException(e);
+            }
+            catch (TaskCanceledException te)
+            {
+                throw new ApiTimeOutException(te);
+            }
+
+            if (response.IsSuccessStatusCode)
+            {
+                var res = await response.Content.ReadAsStringAsync();
+                var res2 = JToken.Parse(res).ToString(Formatting.Indented);
+                this.logger.LogInformation(res2);
+                return res2;
+            }
+
+            return null;
+        }
+
+
+        internal async Task<TResponse> HttpGet<TResponse>(string url, Func<ProblemsContainer, Exception> onError)
+            where TResponse : new()
+        {
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(HttpMethod.Get, url, onError);
+        }
+
+
+        internal async Task<TResponse> HttpPatch
+            <TPayLoad, TResponse>(string url, Func<ProblemsContainer, Exception> onError, TPayLoad payload)
+            where TResponse : new()
+        {
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(new HttpMethod("PATCH"), url, onError, payload);
+        }
+
+
+        internal async Task<TResponse> HttpPost
+            <TPayLoad, TResponse>(string url, Func<ProblemsContainer, Exception> onError, TPayLoad payload)
+            where TResponse : new()
+        {
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(HttpMethod.Post, url, onError, payload);
+        }
+
+
+        internal async Task<T> SendHttpRequestAndProcessHttpResponse
+            <T>(HttpMethod httpMethod, string url, Func<ProblemsContainer, Exception> onError, object payload = null)
+            where T : new()
+        {
+            var requestMessage = new HttpRequestMessage(httpMethod, url);
+
+            return await SendHttpRequestAndProcessHttpResponse<T>(requestMessage, onError, payload);
+        }
+
+
+        internal async Task<T> SendHttpRequestAndProcessHttpResponse
+            <T>(HttpRequestMessage requestMessage, Func<ProblemsContainer, Exception> onError, object payload = null)
+            where T : new()
+        {
+            UpdateRequest(requestMessage, payload);
+
+            HttpResponseMessage response = await this.client.SendAsync(requestMessage, HttpCompletionOption.ResponseHeadersRead).ConfigureAwait(false);
+            
+            if (response.IsSuccessStatusCode)
```

Comment:
Consider adding a try-clause so that if something explodes in here we can still catch it and attach the HTTP response. Something as innocent as parsing JSON may fail.